server/api.go

```diff
@@ -381,6 +374,38 @@ func (a *API) getConnectedChannels(w http.ResponseWriter, r *http.Request) {
 	a.writeJSON(w, http.StatusOK, paginatedLinks)
 }
 
+func (a *API) getMSTeamsTeamList(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get(HeaderMattermostUserID)
+	teams, statusCode, err := a.p.GetMSTeamsTeamList(userID)
+	if err != nil {
+		http.Error(w, "Error occurred while fetching the MS Teams team list.", statusCode)
+		return
+	}
+
+	sort.Slice(teams, func(i, j int) bool {
+		return fmt.Sprintf("%s_%s", teams[i].DisplayName, teams[i].ID) < fmt.Sprintf("%s_%s", teams[j].DisplayName, teams[j].ID)
+	})
+
+	searchTerm := r.URL.Query().Get(QueryParamSearchTerm)
```

Comment:
should we add some minimum length for the search term?

Reply:
discussed